pkg/kubeadm/cmd/join.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+	kubenode "k8s.io/kubernetes/pkg/kubeadm/node"
+	kubeadmutil "k8s.io/kubernetes/pkg/kubeadm/util"
+)
+
+func NewCmdJoin(out io.Writer, params *kubeadmapi.BootstrapParams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "join",
+		Short: "Run this on other servers to join an existing cluster.",
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Yes, I have looked at what kubectl does and started refactoring them, see
init.go

On Wed, 31 Aug 2016, 22:03 Joe Beda, <notifications@github.com> wrote:

> In pkg/kubeadm/cmd/join.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77072907>
> :
>
> > +	"fmt"
> > +	"io"
> > +
> > +	"github.com/renstrom/dedent"
> > +	"github.com/spf13/cobra"
> > +
> > +	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
> > +	kubenode "k8s.io/kubernetes/pkg/kubeadm/node"
> > +	kubeadmutil "k8s.io/kubernetes/pkg/kubeadm/util"
> > +)
> > +
> > +func NewCmdJoin(out io.Writer, params *kubeadmapi.BootstrapParams) *cobra.Command {
> > +	cmd := &cobra.Command{
> > +		Use:   "join",
> > +		Short: "Run this on other servers to join an existing cluster.",
> > +		Run: func(cmd *cobra.Command, args []string) {
>
> This probably deserves to be broken out into its own named function.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/57096ca902c5c666d45df191037d9278972af1fd#r77072907>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS2rrbfnXFD5GR_HcQrrjPd_nEH08ks5qlewKgaJpZM4JqsTH>
> .
>